### PR TITLE
WIP Rename as op and Decorator fix

### PR DESCRIPTION
**Pull request by springcoil**
Based on https://github.com/pymc-devs/pymc3/pull/787 (couldn't push to that branch easily). 

We'll see if this works.

### CODE REVIEW

**Comment by fonnesbeck** [comment]
Seems like a lot of code repetition for just `nonlocal` vs `global`. Can that statement not just be embedded in an `if` statement?

File: pymc3/model.py [hunk 1]
```diff
@@ -720,3 +720,50 @@ def all_continuous(vars):
...
+        itypes = kwds.get('itypes')
+        otypes = kwds.get('otypes') or [t.dvector]
+
+        # Wrapper infers types of inputs
+        @wraps(f)
+        def wrapper(*inputs, **kwinputs):
+            global itypes, otypes
+
+            if not itypes:
+                itypes = [arg.type for arg in inputs]
```
Comment (twiecki) [inline pymc3/model.py]: That's neat!

**Comment by springcoil** [comment]
I will have a look in a few days.

File: pymc3/model.py (hunk 1 above)
Comment (springcoil) [reply pymc3/model.py]: Thanks I think it needs a bit of a refactor as tests are failing. However I think it's just a cosmetic change

**Comment by springcoil** [comment]
Thanks @fonnesbeck I changed code a bit.

File: pymc3/model.py [hunk 2]
```diff
@@ -688,7 +689,7 @@ def __init__(self, type=None, owner=None, index=None, name=None,
             normalRV = transform.backward(self.transformed)
 
             theano.Apply(theano.compile.view_op, inputs=[
-                         normalRV], outputs=[self])
+                normalRV], outputs=[self])
```
Comment (twiecki) [inline pymc3/model.py]: Your editor is doing weird things.

**Comment by springcoil** [comment]
Yeah using PyCharm...
I'm sure I could change the settings.

**Comment by springcoil** [comment]
I've no idea how to get this working... Anyone else have a suggestion?

**Comment by twiecki** [comment]
What's the error?

**Comment by springcoil** [comment]
Syntax errors largely. I'll share later

`global itypes, otypes SyntaxError: name 'otypes' is nonlocal and global`

**Comment by springcoil** [comment]
I've never had time to work on this. So closing

**Comment by springcoil** [comment]
I've never had time to work on this. So closing

**Pull request closed without merging**